src/oscar/apps/address/abstract_models.py

```diff
@@ -332,25 +334,41 @@ def name(self):
 
     # Helpers
 
+    def get_field_values(self, fields):
+        field_values = []
+        for field in fields:
+            # Title is special case
+            if field == 'title':
+                value = self.get_title_display()
+            else:
+                value = getattr(self, field)
+            field_values.append(value)
+        return field_values
+
+    def get_address_field_values(self, fields, include_salutation=True, include_country=True):
+        field_values = self.get_field_values(fields)
+        if include_salutation:
+            field_values = [self.salutation] + self.get_field_values(fields)
+        field_values = [f.strip() for f in field_values if f]
+        if include_country:
+            try:
+                field_values.append(self.country.printable_name)
+            except exceptions.ObjectDoesNotExist:
+                pass
+        return field_values
+
     def generate_hash(self):
         """
         Returns a hash of the address summary
         """
-        # We use an upper-case version of the summary
-        return zlib.crc32(self.summary.strip().upper().encode('UTF8'))
+        field_values = self.get_address_field_values(self.hash_fields)
+        return zlib.crc32(field_values.upper().encode('UTF8'))
```

Comment:
Looks like you need to put this through `join_fields()` before trying to hash it.